openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/IngestionPipelineRepository.java

```diff
@@ -85,19 +85,25 @@ public void storeEntity(IngestionPipeline ingestionPipeline, boolean update) thr
     // Relationships and fields such as href are derived and not stored as part of json
     EntityReference owner = ingestionPipeline.getOwner();
     EntityReference service = ingestionPipeline.getService();
+    OpenMetadataConnection openmetadataConnection = ingestionPipeline.getOpenMetadataServerConnection();
 
     SecretsManager secretsManager = SecretsManagerFactory.getSecretsManager();
+
     if (secretsManager != null) {
       secretsManager.encryptOrDecryptIngestionPipeline(ingestionPipeline, true);
+      // We store the OM sensitive values in SM separately
+      openmetadataConnection =
+          secretsManager.encryptOrDecryptOpenMetadataConnection(openmetadataConnection, true, true);
     }
 
     // Don't store owner. Build it on the fly based on relationships
-    ingestionPipeline.withOwner(null).withService(null).withHref(null);
+    // We don't want to store the OM connection.
+    ingestionPipeline.withOwner(null).withService(null).withHref(null).withOpenMetadataServerConnection(null);
```

Comment:
Would we be able to use decryptorNullify here?
sorry for asking the above question as not sure what is the main func of that, but based on name, i think it might be able to set children objects to null

Reply:
`decryptorNullify` is a method we use in the resources before sending the responses from the API to decrypt the values stored in DB, which are encrypted. However, in the repository, we set those values to null since the relationship with those Objects is built on the fly based on the relationships.